### PR TITLE
Improve iOS 10.3 review dialog support + documentation

### PR DESCRIPTION
**Pull request by MartinMoizard**
Hello,

This is a fork of https://github.com/UrbanApps/Armchair/pull/102 + I added the requested documentation.

I went ahead and did that because it only missed the documentation, and I think we can all benefit from this nice change.

@coneybeare let me know if I should change anything more.

Thanks

### CODE REVIEW

File: Source/Armchair.swift [hunk 1]
```diff
@@ -1703,23 +1737,19 @@ open class Manager : ArmchairManager {
     #endif
     
     private func hideRatingAlert() {
+        #if os(OSX)
```
Comment (coneybeare) [inline Source/Armchair.swift]: Why was all this edited? It still needs to be split by iOS and OSX to properly handle Mac App implementations.

**Comment by coneybeare** [comment]
Alright, looks good, thanks!

**Comment by coneybeare** [comment]
Just checked and you don't compile for Mac! please fix so that the Mac Example App compiles and works. Remember that SKStoreReviewController and store kit reviewing is not available for Mac, so all that logic needs to be contained in a compile-time `#if os(iOS)` block.

**Comment by ianbytchek** [comment]
@MartinMoizard 👋 can you look into macOS demo issues?

**Pull request merged**